content/blog/posts/o3de-motion-matching-2022-04.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Motion Matching in O3DE, a Data-Driven Animation Technique"
+date: 2022-05-12
+slug: blog-motionmatching
+author: Benjamin Jillich
+blog_img: "/images/blog/motion-matching/blog_img.png"
+full_img: "/images/blog/motion-matching/full_image.png"
+---
+
+Motion matching is a data-driven animation technique that is gaining popularity since games like 'The Last Of Us Part II', 'FIFA', and 'Half-Life: Alyx' have adotped it. Users have higher expectations for animation quality including more realistic animations and a greater variety of interactions between characters and their environments. This growth in expectation resulted in increasingly complex animation graphs that became harder to maintain and more expensive to create. Motion matching is a way to animate a character in a controlled way by jumping and blending between pre-recorded animation sequences several times a second, synthesizing the animation for the character. It can be used for realistic locomotion without the need to manually create carefully timed transition points and synchronization between animation clips.
+
+We brought motion matching to O3DE's animation system [EMotion FX](https://www.o3de.org/docs/user-guide/visualization/animation/char-intro/) as a blend tree node so that it can still be used and mixed with everything else available in the animation graph.
+
+{{< video src="/images/blog/motion-matching/controlled-character.mp4" autoplay="true" loop="true" width="100%" info="Gamepad controlled character using motion matching." >}}
+<br>
+<br>
+
+## Motion Database
+
+To animate a character using motion matching, we need a large library of character animations such as walking slowly, running fast, narrow and wide turns, accelerating and decelerating, and many more. A large variety of movements are required so that there is a piece of animation for every situation. Kristjan Zadziuk and his team ([GDC 2016 Talk, Animation Bootcamp: Motion Matching](https://www.youtube.com/watch?v=_Bd2T7uP9VA)) gave this some structure and worked out a set of *dance cards*. These dance cards describe paths actors move along in a motion capture studio that capture the full range of motions for best visual results. They also minimize variations and redundant data that decrease the algorithm's performance. You can see a sample database of moves in the [O3DE repository](https://github.com/o3de/o3de/tree/development/Gems/MotionMatching/Assets/Animations).
+
+## Motion Features
+
+A smooth animation is the result of continuity. Sudden changes in position or velocity will result in unrealistic motion. We want to synthesize an animation that is smooth and matches the intention of the user's interaction. If the user presses forward on the game-pad, it is not enough to transition to an animation sequence that results in a smooth animation while the expected direction of movement is ignored.
```

Comment:
nit: We use "smooth" a few times in this paragraph and below. May want to consider alternatives just to avoid repetition.

Reply:
Changed